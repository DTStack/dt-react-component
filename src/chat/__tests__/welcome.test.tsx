import React from 'react';
import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom';

import Welcome, { type IWelcomeProps } from '../welcome';

describe('Test Welcome Component', () => {
    beforeEach(() => cleanup());

    const defaultProps: IWelcomeProps = {
        title: 'Welcome Title',
        description: 'This is a description',
    };

    it('matches the snapshot with default props', () => {
        const { asFragment } = render(<Welcome {...defaultProps} />);
        expect(asFragment()).toMatchSnapshot('default welcome');
    });

    it('renders with a custom icon', () => {
        const customIcon = <div data-testid="custom-icon">Custom Icon</div>;
        const { getByTestId } = render(<Welcome {...defaultProps} icon={customIcon} />);
        expect(getByTestId('custom-icon')).toBeInTheDocument();
    });

    it('applies custom className and style', () => {
        const { container } = render(
            <Welcome
                {...defaultProps}
                className="custom-class"
                style={{ backgroundColor: 'red' }}
            />
        );
        expect(container.firstChild).toHaveClass('custom-class');
        expect(container.firstChild).toHaveStyle('background-color: red');
    });

    it('renders title and description correctly', () => {
        const { getByText } = render(<Welcome {...defaultProps} />);
        expect(getByText('Welcome Title')).toBeInTheDocument();
        expect(getByText('This is a description')).toBeInTheDocument();
    });

    it('renders without description when not provided', () => {
        const { queryByText } = render(<Welcome title="Title Only" />);
        expect(queryByText('Title Only')).toBeInTheDocument();
        expect(queryByText('This is a description')).not.toBeInTheDocument();
    });
});
