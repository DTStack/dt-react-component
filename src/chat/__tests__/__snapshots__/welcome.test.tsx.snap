// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Welcome Component matches the snapshot with default props: default welcome 1`] = `
<DocumentFragment>
  <div
    class="dtc__welcome"
  >
    <div
      class="dtc__welcome__content"
    >
      <section
        class="dtc-flex"
        style="flex-wrap: nowrap; justify-content: normal; align-items: center; gap: 8px;"
      >
        <span
          class="dtc__icon"
          style="font-size: 24px;"
        >
          <svg
            fill="none"
            height="1em"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-path="url(#clip0_3878_7498)"
              id="AiAvatar"
            >
              <path
                d="M1.664 4.428c-1.896-2.292 1.08-5.1 3.18-2.172.324.58.87 1.584 1.074 2.292-.72.492-1.536.972-2.136 1.392-.358-1.084-1.116-.907-1.71-1.192a1.2 1.2 0 0 1-.408-.32"
                fill="#F2F7FA"
                id="Vector 13"
              />
              <path
                clip-rule="evenodd"
                d="M2.791 4.502a8 8 0 0 1 .228-.285c.282.082.61.074.886-.058a.95.95 0 0 0 .497-.555c.06-.18.07-.375.037-.577.064-.054.136-.104.228-.158.251.463.537 1.024.716 1.496-.26.171-.527.341-.792.51l-.027.016c-.203.129-.405.257-.598.383a1.64 1.64 0 0 0-.78-.646 3 3 0 0 0-.395-.126m1.365-2.376c-.2.124-.382.266-.558.478a.45.45 0 0 0-.082.427c.058.18.04.263.033.284-.004.012-.008.02-.032.032a.33.33 0 0 1-.163.021.4.4 0 0 1-.109-.023.47.47 0 0 0-.404-.108.5.5 0 0 0-.127.045.8.8 0 0 0-.19.144 4 4 0 0 0-.292.33 10 10 0 0 0-.26.338c-.75-.949-.489-1.91.066-2.36.286-.23.66-.34 1.065-.256.317.066.683.256 1.053.648M3.285.597c.682.14 1.35.597 1.924 1.396l.015.021.013.022c.317.569.893 1.62 1.114 2.387l.09.313-.269.183c-.357.244-.736.485-1.094.712l-.032.02c-.37.235-.716.455-1.006.658l-.495.346-.19-.574a1.16 1.16 0 0 0-.237-.438.74.74 0 0 0-.273-.182 3 3 0 0 0-.385-.115l-.03-.007a3 3 0 0 1-.553-.185l.195-.406-.195.406a1.6 1.6 0 0 1-.56-.44c-1.099-1.328-.82-2.89.155-3.68A2.16 2.16 0 0 1 3.285.597"
                fill="#64698B"
                fill-rule="evenodd"
                id="Vector 13 (Stroke)"
              />
              <path
                d="M18.465 4.428c1.896-2.292-1.08-5.1-3.18-2.172-.324.58-.87 1.584-1.075 2.292.72.492 1.536.972 2.136 1.392.358-1.084 1.116-.907 1.711-1.192a1.2 1.2 0 0 0 .408-.32"
                fill="#F2F7FA"
                id="Vector 13_2"
              />
              <path
                clip-rule="evenodd"
                d="M17.338 4.502a8 8 0 0 0-.228-.285c-.282.082-.61.074-.886-.058a.95.95 0 0 1-.497-.555 1.2 1.2 0 0 1-.037-.577 1.7 1.7 0 0 0-.229-.158c-.25.463-.536 1.024-.715 1.496.26.171.527.341.792.51l.026.016q.308.194.599.383.086-.134.189-.244c.184-.198.39-.32.59-.402.135-.055.273-.095.396-.126m-1.366-2.376c.2.124.383.266.559.478a.45.45 0 0 1 .082.427c-.058.18-.04.263-.033.284.004.012.008.02.032.032a.33.33 0 0 0 .163.021.4.4 0 0 0 .109-.023.47.47 0 0 1 .404-.108.5.5 0 0 1 .127.045.8.8 0 0 1 .19.144c.087.085.192.207.292.33.084.105.175.222.26.338.75-.949.488-1.91-.066-2.36-.286-.23-.66-.34-1.065-.256-.317.066-.683.256-1.054.648m.986 1.927.003.002zM16.843.597c-.681.14-1.35.597-1.924 1.396l-.014.021-.013.022c-.317.569-.893 1.62-1.114 2.387l-.09.313.268.183c.357.244.737.485 1.095.712l.031.02c.371.235.716.455 1.006.658l.496.346.19-.574c.073-.222.158-.354.236-.438a.74.74 0 0 1 .274-.182c.11-.046.229-.076.385-.115l.03-.007c.159-.04.358-.092.552-.185l-.194-.406.194.406c.197-.094.389-.231.56-.44 1.1-1.328.82-2.89-.154-3.68a2.16 2.16 0 0 0-1.814-.437"
                fill="#64698B"
                fill-rule="evenodd"
                id="Vector 13 (Stroke)_2"
              />
              <path
                d="M18.782 11.28c1.332 7.656-3.852 8.124-8.736 8.124s-9.883-.588-8.928-7.5c.804-5.82 4.044-8.268 8.928-8.268s7.82 2.376 8.736 7.644"
                fill="#F2F7FA"
                id="Ellipse 1"
              />
              <path
                clip-rule="evenodd"
                d="M4.35 5.925c-1.417 1.247-2.394 3.201-2.786 6.04-.232 1.678-.096 2.927.277 3.86.37.924.985 1.571 1.777 2.032 1.622.942 3.973 1.097 6.428 1.097 1.218 0 2.43-.03 3.55-.189s2.116-.444 2.913-.935c.786-.485 1.39-1.179 1.73-2.194.344-1.026.425-2.41.1-4.28-.447-2.567-1.375-4.369-2.73-5.532s-3.193-1.738-5.563-1.738c-2.368 0-4.281.592-5.697 1.839m-.596-.675c1.627-1.433 3.776-2.064 6.292-2.064 2.514 0 4.585.612 6.149 1.955 1.562 1.341 2.56 3.361 3.03 6.062.341 1.958.274 3.504-.132 4.719a4.86 4.86 0 0 1-2.111 2.674c-.944.582-2.076.892-3.26 1.06-1.185.169-2.451.198-3.676.198-2.43 0-5.02-.14-6.88-1.22-.948-.55-1.709-1.345-2.16-2.475-.449-1.122-.58-2.538-.334-4.317.412-2.98 1.457-5.16 3.082-6.592"
                fill="#64698B"
                fill-rule="evenodd"
                id="Ellipse 1 (Stroke)"
              />
              <g
                filter="url(#filter0_d_3878_7498)"
                id="Union"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.534 16.853q.162 0 .318-.014c.148-.013.295-.027.443-.027h7.769a.002.002 0 0 0 .002-.003q0-.001.003-.002.193.022.393.022c1.903 0 3.445-1.526 3.445-3.408s-1.427-3.3-2.82-2.928l-.08.021c-.204.055-.444.119-.704.196-.608.18-1.276.075-1.76-.334a2 2 0 0 0-.258-.18l-.853-.486a2.88 2.88 0 0 0-2.864.006l-.835.48q-.135.08-.253.18a1.77 1.77 0 0 1-1.632.342q-.42-.119-.89-.242c-1.393-.373-2.868 1.087-2.868 2.97 0 1.881 1.542 3.407 3.444 3.407"
                  fill="url(#paint0_linear_3878_7498)"
                  fill-rule="evenodd"
                />
              </g>
              <g
                filter="url(#filter1_d_3878_7498)"
                id="Union (Stroke)"
              >
                <path
                  clip-rule="evenodd"
                  d="M5.534 16.153q.129 0 .255-.011l.02-.002c.138-.012.31-.028.486-.028h7.69a.7.7 0 0 1 .162 0q.154.017.316.017c1.522 0 2.743-1.22 2.743-2.708 0-.767-.29-1.414-.687-1.82s-.853-.538-1.252-.432l-.08.022c-.204.054-.435.116-.685.19-.804.238-1.724.11-2.41-.47q-.073-.06-.154-.107l-.853-.486a2.18 2.18 0 0 0-2.168.005l-.834.48a1 1 0 0 0-.147.104c-.638.548-1.501.703-2.278.485a70 70 0 0 0-.88-.24c-.393-.105-.861.03-1.276.453-.412.42-.712 1.079-.712 1.84 0 1.489 1.222 2.708 2.744 2.708m8.53.659a.002.002 0 0 0 .002-.003q0-.001.003-.002l.015.002a3.5 3.5 0 0 0 .682.007c1.76-.152 3.14-1.614 3.14-3.395 0-1.882-1.426-3.3-2.82-2.928l-.08.021c-.203.055-.443.119-.703.196-.608.18-1.276.075-1.76-.334a2 2 0 0 0-.258-.18l-.853-.486a2.88 2.88 0 0 0-2.864.006l-.835.48q-.135.08-.253.18a1.77 1.77 0 0 1-1.632.342q-.42-.119-.89-.242c-1.393-.373-2.868 1.087-2.868 2.97 0 1.881 1.542 3.407 3.444 3.407q.162 0 .318-.014l.012-.001c.144-.013.287-.026.431-.026z"
                  fill="url(#paint1_linear_3878_7498)"
                  fill-rule="evenodd"
                />
              </g>
              <path
                clip-rule="evenodd"
                d="M3.906 12.127c.377-.328.863-.43 1.196-.427a.2.2 0 1 1-.003.4c-.265-.002-.647.082-.93.329-.272.236-.483.645-.377 1.362a.2.2 0 1 1-.395.059c-.123-.831.12-1.384.51-1.723"
                fill="#fff"
                fill-rule="evenodd"
                id="Vector 17 (Stroke)"
              />
              <g
                filter="url(#filter2_d_3878_7498)"
                id="Vector 18 (Stroke)"
              >
                <path
                  clip-rule="evenodd"
                  d="M13.12 12.25a.75.75 0 0 1 .75.75v.936a.75.75 0 0 1-1.5 0V13a.75.75 0 0 1 .75-.75"
                  fill="#fff"
                  fill-rule="evenodd"
                />
              </g>
              <g
                filter="url(#filter3_d_3878_7498)"
                id="Vector 19 (Stroke)"
              >
                <path
                  clip-rule="evenodd"
                  d="M6.88 12.25a.75.75 0 0 1 .75.75v.936a.75.75 0 0 1-1.5 0V13a.75.75 0 0 1 .75-.75"
                  fill="#fff"
                  fill-rule="evenodd"
                />
              </g>
            </g>
            <defs>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="9.922"
                id="filter0_d_3878_7498"
                width="18.216"
                x="0.89"
                y="8.132"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset />
                <fegaussianblur
                  stdDeviation="0.6"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  values="0 0 0 0 0.233659 0 0 0 0 0.448234 0 0 0 0 1 0 0 0 0.2 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  result="effect1_dropShadow_3878_7498"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_3878_7498"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="9.922"
                id="filter1_d_3878_7498"
                width="18.216"
                x="0.89"
                y="8.132"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset />
                <fegaussianblur
                  stdDeviation="0.6"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  values="0 0 0 0 0.233659 0 0 0 0 0.448234 0 0 0 0 1 0 0 0 0.2 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  result="effect1_dropShadow_3878_7498"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_3878_7498"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="3.876"
                id="filter2_d_3878_7498"
                width="2.94"
                x="11.65"
                y="11.53"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset />
                <fegaussianblur
                  stdDeviation="0.36"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  values="0 0 0 0 1 0 0 0 0 1 0 0 0 0 1 0 0 0 0.8 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  result="effect1_dropShadow_3878_7498"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_3878_7498"
                  result="shape"
                />
              </filter>
              <filter
                color-interpolation-filters="sRGB"
                filterUnits="userSpaceOnUse"
                height="3.876"
                id="filter3_d_3878_7498"
                width="2.94"
                x="5.41"
                y="11.53"
              >
                <feflood
                  flood-opacity="0"
                  result="BackgroundImageFix"
                />
                <fecolormatrix
                  in="SourceAlpha"
                  result="hardAlpha"
                  values="0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 127 0"
                />
                <feoffset />
                <fegaussianblur
                  stdDeviation="0.36"
                />
                <fecomposite
                  in2="hardAlpha"
                  operator="out"
                />
                <fecolormatrix
                  values="0 0 0 0 1 0 0 0 0 1 0 0 0 0 1 0 0 0 0.8 0"
                />
                <feblend
                  in2="BackgroundImageFix"
                  result="effect1_dropShadow_3878_7498"
                />
                <feblend
                  in="SourceGraphic"
                  in2="effect1_dropShadow_3878_7498"
                  result="shape"
                />
              </filter>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="paint0_linear_3878_7498"
                x1="2.739"
                x2="15.115"
                y1="10.208"
                y2="18.018"
              >
                <stop
                  offset="0.025"
                  stop-color="#EBFEFF"
                />
                <stop
                  offset="0.339"
                  stop-color="#8CCFFF"
                />
                <stop
                  offset="0.708"
                  stop-color="#B399FF"
                />
                <stop
                  offset="1"
                  stop-color="#BDA6FF"
                />
              </lineargradient>
              <lineargradient
                gradientUnits="userSpaceOnUse"
                id="paint1_linear_3878_7498"
                x1="2.09"
                x2="15.765"
                y1="9.332"
                y2="19.683"
              >
                <stop
                  stop-color="#00BAC6"
                />
                <stop
                  offset="0.5"
                  stop-color="#0067FF"
                />
                <stop
                  offset="1"
                  stop-color="#450FDE"
                />
              </lineargradient>
              <clippath
                id="clip0_3878_7498"
              >
                <path
                  d="M0 0h20v20H0z"
                  fill="#fff"
                />
              </clippath>
            </defs>
          </svg>
        </span>
        <span
          class="dtc__welcome__title"
        >
          Welcome Title
        </span>
      </section>
      <div
        class="dtc__welcome__description"
      >
        This is a description
      </div>
    </div>
  </div>
</DocumentFragment>
`;
